libraries/ts-command-line/src/providers/TabCompletionAction.ts

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import stringArgv from 'string-argv';
+
+import { CommandLineIntegerParameter } from '../parameters/CommandLineIntegerParameter';
+import { CommandLineStringParameter } from '../parameters/CommandLineStringParameter';
+import { CommandLineParameterKind, CommandLineParameter } from '../parameters/BaseClasses';
+import { CommandLineAction } from './CommandLineAction';
+import { CommandLineChoiceParameter } from '..';
+import { CommandLineConstants } from '../Constants';
+
+interface IParameter {
+  name: string;
+  parameterInfo: CommandLineParameter;
+}
+
+const DEFAULT_WORD_TO_AUTOCOMPLETE: string = '';
+const DEFAULT_POSITION: number = 0;
+
+export class TabCompleteAction extends CommandLineAction {
```

Comment:
Can/do we exclude this action from what's printed when someone invokes the command with the `--help` argument?

Reply:
We can, but why?

Reply:
It's not something someone would ever explicitly run, so it's only confusing to include it in the help message.

Reply:
Okay. I just checked if the dotnet core CLI shows its tab-complete command in its help. It doesn't either. I will make the change.

Reply:
Our help text is printed using `ArgumentParser.formatHelp()`. Excluding this action does not seem to be straightforward.

Reply:
Let's fix this in a separate PR.